Remove pip --build-option and %py_setup_args

mcepl opened the pull request:
--build-option and --global-option are deprecated. A possible replacement is to use --config-settings. Discussion can be found at gh#pypa/pip#11859

danigm reviewed (approved):
LGTM

bnavigator commented:
See #165!

https://github.com/pypa/pip/issues/11859 and https://github.com/pypa/setuptools/issues/3896 are **still** not closed. Without checking, I suspect that this change will cause some breakage of packages in Factory depending on the deprecated setuptools behavior.

mcepl commented:
Perhaps we don’t need to use `%py_setup_args` at all. See [this revision of `python-pyo`](https://build.opensuse.org/package/rdiff/devel:languages:python:numeric/python-pyo?linkrev=base&rev=24), where I use happily `$PIP_CONFIG_SETTINGS`, which seems like more pleasant solution than injecting some macro to the command call.

bnavigator commented:
The good news is, that py_setup_args is used nowhere in Factory.

https://github.com/search?q=repo%3Abmwiedemann%2FopenSUSE%20py_setup_args&type=code

mcepl commented:
> The good news is, that py_setup_args is used nowhere in Factory.

So, instead of this PR, should I prepare instead removal of `py_setup_args` altogether?

bnavigator commented:
No objection from me

danigm commented:
> > The good news is, that py_setup_args is used nowhere in Factory.
> 
> So, instead of this PR, should I prepare instead removal of `py_setup_args` altogether?

Good idea. And we can just recommend to use `PIP_CONFIG_SETTINGS` env var when needed.

mcepl committed:
Remove py_setup_args macro completely.

The functionality is fully equivalent to setting the
environmental variable $PIP_CONFIG_SETTINGS and we don’t need to
complicate our macros with it.

See for example https://is.gd/yFaA8q

mcepl commented:
@bnavigator Whatever is the result of https://github.com/pypa/pip/issues/11859 and https://github.com/pypa/setuptools/issues/3896 this current PR doesn’t change anything on the result.

bnavigator reviewed (approved):
LGTM
